### PR TITLE
Require valid names/ids in data passed to OfferEvaluator/StateStore

### PR DESCRIPTION
**Pull request by nickbp**
- TaskInfo requirements:
  - must have a name
  - if task id is non-empty, it must align with the name
  - if executor is provided, it must have a name (only at StateStore)
- ExecutorInfo requirements:
  - must have a name
  - if executor id is non-empty, it must align with the name
- TaskStatus requirements (only at StateStore):
  - must have a valid/parseable task id
  - must have a valid/parseable executor id

### CODE REVIEW

File: src/main/java/org/apache/mesos/offer/ExecutorRequirement.java [hunk 1]
```diff
@@ -2,56 +2,89 @@
...
+                        executorInfo, e));
+            }
+            if (!executorName.equals(executorInfo.getName())) {
+                throw new InvalidRequirementException(String.format(
+                        "When non-empty, ExecInfo.id must align with ExecInfo.name. Use "
+                        + "ExecutorUtils.toExecutorId(): %s", executorInfo));
+            }
+        }
+        // ExecutorID is always overwritten with a new UUID, even if already present:
+        return ExecutorInfo.newBuilder(executorInfo)
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/offer/ExecutorRequirement.java]: I would separate validation from generation of the new ExecutorID

File: src/main/java/org/apache/mesos/offer/TaskRequirement.java [hunk 2]
```diff
@@ -1,62 +1,81 @@
...
+                        + "TaskUtils.toTaskId(): %s", taskInfo));
+            }
+        }
+        if (taskInfo.hasExecutor()) {
+            throw new InvalidRequirementException(String.format(
+                    "TaskInfo must not contain ExecutorInfo. "
+                    + "Use ExecutorRequirement for any Executor requirements: %s", taskInfo));
+        }
+        // TaskID is always overwritten with a new UUID, even if already present:
+        return TaskInfo.newBuilder(taskInfo)
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/offer/TaskRequirement.java]: Same here.  Validation and construction should be separate.

**Comment by gabrielhartmann** [comment]
@nickbp: Other than those 2 comments, looks good to me.

File: src/main/java/org/apache/mesos/offer/ExecutorRequirement.java (hunk 1 above)
Comment (nickbp) [inline src/main/java/org/apache/mesos/offer/ExecutorRequirement.java]: Done

File: src/main/java/org/apache/mesos/offer/TaskRequirement.java (hunk 2 above)
Comment (nickbp) [inline src/main/java/org/apache/mesos/offer/TaskRequirement.java]: Done

**Pull request merged**